internal/x11/win/frame.go

```diff
@@ -579,17 +585,40 @@ func (f *frame) mouseRelease(x, y int16) {
 		return
 	}
 
-	obj := wm.FindObjectAtPixelPositionMatching(int(relX), int(relY), f.canvas,
-		func(obj fyne.CanvasObject) bool {
-			_, ok := obj.(fyne.Tappable)
-			return ok
-		},
-	)
-
-	if obj == nil {
+	if f.ctx != nil && f.cancelFunc != nil {
+		f.doubleClick = true
+		f.cancelFunc()
 		return
 	}
-	obj.(fyne.Tappable).Tapped(&fyne.PointEvent{})
+	go func() {
+		f.ctx, f.cancelFunc = context.WithDeadline(context.TODO(), time.Now().Add(time.Millisecond*300))
+		defer f.cancelFunc()
+		select {
+		case <-f.ctx.Done():
+			if f.doubleClick == true {
+				if f.client.Maximized() {
```

Comment:
This is putting border logic back into the X11 package which is what I've been trying to avoid.
Instead it would be cleaner to put the double tap /handling/ into the UI, and here just trigger double tap handlers - i.e. the `fyne.DoubleTappable` interface.